pulsar-functions/instance/src/main/python/contextimpl.py

```diff
@@ -133,7 +133,7 @@ def get_output_topic(self):
   def get_output_serde_class_name(self):
     return self.instance_config.function_details.outputSerdeClassName
 
-  def publish(self, topic_name, message, serde_class_name="serde.IdentitySerDe", properties=None, compression_type=None):
+  def publish(self, topic_name, message, serde_class_name="serde.IdentitySerDe", properties=None, compression_type=None, callback=None):
```

Comment:
seems this change is resolving a different problem, no?

If so, can you update the issue description? or separate it into a different PR?

Reply:
not really since if we don't provide the user to be able to ack on a fail when publishing, the a deadlock with consumers can happen as well